das_api/src/api/api_impl.rs

```diff
@@ -363,4 +364,34 @@ impl ApiContract for DasApi {
             group_size: gs.size,
         })
     }
+
+    async fn get_signatures_for_asset(
+        self: &DasApi,
+        payload: GetSignaturesForAsset,
+    ) -> Result<TransactionSignatureList, DasApiError> {
+        let GetSignaturesForAsset { 
+            id,
+            limit,
+            page,
+            before,
+            after,
```

Comment:
We could. We'd need a different keyset though, since it's based on the Seq and not the ID or createdAt field.

But page-based is reasonable for this API since the volume size isn't too large per key. It will iterate quickly enough over 1M records, and it shouldn't get much bigger than that.